storage-proofs-update/src/challenges.rs

```diff
@@ -33,15 +38,19 @@ pub struct Challenges {
     challenges_remaining: usize,
 }
 
-impl Challenges {
-    pub fn new(sector_nodes: usize, comm_r_new: TreeRDomain, k: usize) -> Self {
+impl<F> Challenges<F>
+where
+    F: PrimeFieldBits,
+    TreeRDomain<F>: Domain<Field = F>,
+{
+    pub fn new(sector_nodes: usize, comm_r_new: TreeRDomain<F>, k: usize) -> Self {
         let partitions = partition_count(sector_nodes);
         assert!(k < partitions);
 
         let challenge_bit_len = sector_nodes.trailing_zeros() as usize;
         let partition_bit_len = partitions.trailing_zeros() as usize;
         let random_bits_per_challenge = challenge_bit_len - partition_bit_len;
-        let challenges_per_digest = Fr::CAPACITY as usize / random_bits_per_challenge;
+        let challenges_per_digest = F::CAPACITY as usize / random_bits_per_challenge;
```

Comment:
Note: `F::Capacity` and `F::NumBits` are equal for all three of the valid fields.